tests/unittests/test_cli.py

```diff
@@ -218,18 +219,27 @@ def test_wb_devel_schema_subcommand_parser(self):
             self.stderr.getvalue())
 
     def test_wb_devel_schema_subcommand_doc_content(self):
-        """Validate that doc content is sane from known examples."""
+        """Validate that doc content is sane from known examples.
+
+        Ensure that schema doc is returned
+        """
+
+        # Note: patchStdoutAndStderr() is convenient for reducing boilerplate,
```

Comment:
Ultimately, we'd like to move away from subclassing test_helpers.FilesystemMockingTestCase which is a subclass of unittest.TestCase derivatives where possible and prefer to use pytest framework for discovering any tests prefaced with "Test". This gives the ability to use capsys fixture. I'm not certain this particular case is a candidate for shifting aware from unittest.TestCase for the other unit tests in this test_cli module. But I can see that the particular test case is an easy shift away from the FileSystemMocking test case. Your call on what you'd prefer to do on this PR. It might be more work than it's worth for this iteration to restructure the tests in this module.

```diff
diff --git a/tests/unittests/test_cli.py b/tests/unittests/test_cli.py
index d6909e7a..64e7b8cb 100644
--- a/tests/unittests/test_cli.py
+++ b/tests/unittests/test_cli.py
@@ -282,4 +282,31 @@ class TestCLI(test_helpers.FilesystemMockingTestCase):
         self.assertFalse(parseargs.debug)
         self.assertFalse(parseargs.force)
 
+
+class TestUsePyTest:
+    def _call_main(self, sysv_args=None):
+        if not sysv_args:
+            sysv_args = ['cloud-init']
+        try:
+            return cli.main(sysv_args=sysv_args)
+        except SystemExit as e:
+            return e.code
+
+    def test_wb_devel_schema_subcommand_doc_content(self, capsys):
+        """Validate that doc content is sane from known examples.
+        Ensure that schema doc is returned
+        """
+        self._call_main(['cloud-init', 'devel', 'schema', '--docs', 'all'])
+        expected_doc_sections = [
+                '**Supported distros:** all',
+                ('**Supported distros:** almalinux, alpine, centos, '
+                    'cloudlinux, debian, eurolinux, fedora, openEuler, '
+                    'opensuse, photon, rhel, rocky, sles, ubuntu, virtuozzo'),
+                '**Config schema**:\n    **resize_rootfs:** '
+                '(true/false/noblock)',
+                '**Examples**::\n\n    runcmd:\n        - [ ls, -l, / ]\n'
+        ]
+        stdout, stderr = capsys.readouterr()
+        for expected in expected_doc_sections:
+            assert expected in stdout
 # : ts=4 expandtab
```

Reply:
I haven't used capsys directly before, but it looks like `pytest --capture=tee-sys` with capsys would accomplish most of what I was trying to do. This test in particular produces a large amount of output, so the contextmanager allowed me to programatically filter values before printing captured output for debugging, which I don't see as an option with capsys (however maybe I just missed that in the docs). For most use cases duplicating output to stdout/stderr would be more than enough.

I am happy to separate this case out using the suggestion above, but given the size of this PR already, I think I'd prefer to make a `s/FileSystemMocking/capsys/g` conversion of this file into a separate PR.

Reply:
+1 this can wait. not needed for this branch.